test/creator.cpp

```diff
@@ -184,7 +185,11 @@ TEST(ZimCreator, createZim)
 #if defined(ENABLE_XAPIAN)
   direntOffset = offset_t(reader->read_uint<offset_type>(offset_t(urlPtrPos + 24)));
   dirent = direntReader.readDirent(direntOffset);
-  test_article_dirent(dirent, 'X', "title/xapian", "title/xapian", xapian_mimetype, cluster_index_t(1), blob_index_t(0));
+  test_article_dirent(dirent, 'X', "fulltext/xapian", "fulltext/xapian", xapian_mimetype, cluster_index_t(1), Filter<blob_index_t>());
```

Comment:
I've introduce the `NoneType` and `None` object. This is simpler.
It is not a problem we hide the type as we are not checking the (nonexistent) value.

I've also add the check method to the Optional to avoid repeat the `if(filter.active) ...`.